gobblin-config-management/src/main/java/gobblin/config/configstore/ConfigStoreWithImportedBy.java

```diff
@@ -0,0 +1,33 @@
+package gobblin.config.configstore;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+
+import com.typesafe.config.Config;
+
+
+/**
+ * ConfigStoreWithImportedBy indicate this {@ConfigStore} support mapping from the URI to it's imported by URIs
+ * @author mitu
+ *
+ */
+public interface ConfigStoreWithImportedBy extends ConfigStore {
+
+  /**
+   * @param uri - the uri relative to this configuration store
+   * @param version - specify the configuration version in the configuration store.
+   * @return - The {@java.util.Collection} of the URI. Each URI in the collection directly import input uri 
+   *  against input configuration version
+   */
+  public Collection<URI> getImportedBy(URI uri, String version);
+
+  /**
+   * @param uri - the uri relative to this configuration store
+   * @param version - specify the configuration version in the configuration store.
+   * @return - The {@java.util.Map}. The key of the Map is the URI directly import input uri 
+   *  against input configuration version. The value of the Map the directly specified configuration in 
+   *  com.typesafe.config.Config format for corresponding key
+   */
+  public Map<URI, Config> getConfigsImportedBy(URI uri, String version);
```

Comment:
Does this method return own configs or resolved configs? I don't think this method should be in this interface. Instead, I think we need a couple more methods for batch fetches in ConfigStore and ConfigStoreWithResolution.